Sources/SwiftParser/Declarations.swift

```diff
@@ -1697,12 +1697,55 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclIntroducer {
+    var unexpectedBeforeFixity: RawUnexpectedNodesSyntax?
+    var fixity: RawTokenSyntax
+    var unexpectedBeforeOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclIntroducer(_ attrs: DeclAttributes, _ handle: RecoveryConsumptionHandle) -> OperatorDeclIntroducer {
+
+    let fixityText = [Keyword.prefix, .infix, .postfix].map({ $0.defaultText })
+
+    var unexpectedBeforeFixity = RawUnexpectedNodesSyntax(combining: attrs.modifiers?.elements.compactMap { $0.unexpectedBeforeName }, RawUnexpectedNodesSyntax(attrs.attributes?.elements ?? [], arena: self.arena), arena: self.arena)
```

Comment:
> The following should give you RawUnexpectedNodesSyntax? which conforms to UnexpectedNodesCombinable

The attributes in `attrs` is actually optional, so we could not just initialize a `RawUnexpectedNodesSyntax` with it, since that requires a non-optional value.

> Any modifiers after the fixity get added to leftModifiers

I think we could just make modifiers after the fixity parsed into `unexpectedBeforeOperatorKeyword`. It should make it easier.